doc/specs/vulkan/chapters/memory.txt

```diff
@@ -774,29 +790,30 @@ same memory object.
 When memory is allocated, its contents are undefined.
 
 There is an implementation-dependent maximum number of memory allocations
-which can: be simultaneously created on a device.
+that can: be simultaneously created on a device.
 This is specified by the
 <<features-limits-maxMemoryAllocationCount,pname:maxMemoryAllocationCount>>
-member of the sname:VkPhysicalDeviceLimits structure.
+member of the slink:VkPhysicalDeviceLimits structure.
 If pname:maxMemoryAllocationCount is exceeded, fname:vkAllocateMemory will
 return ename:VK_ERROR_TOO_MANY_OBJECTS.
 
-[NOTE]
-.Note
-====
 Some platforms may: have a limit on the maximum size of a single allocation.
 For example, certain systems may: fail to create allocations with a size
 greater than or equal to 4GB.
 Such a limit is implementation-dependent, and if such a failure occurs then
-the error ename:VK_ERROR_OUT_OF_DEVICE_MEMORY should: be returned.
-====
+the error ename:VK_ERROR_OUT_OF_DEVICE_MEMORY must: be returned.
 
 .Valid Usage
 ****
-  * [[VUID-vkAllocateMemory-device-00636]]
-    The number of currently valid memory objects, allocated from
-    pname:device, must: be less than
-    sname:VkPhysicalDeviceLimits::pname:maxMemoryAllocationCount
+  * pname:pAllocateInfo\->pname:allocationSize must: be less than or equal
+    to
+    slink:VkPhysicalDeviceMemoryProperties::pname:memoryHeaps[pname:pAllocateInfo\->pname:memoryTypeIndex].pname:size
+    as returned by flink:vkGetPhysicalDeviceMemoryProperties for the
+    slink:VkPhysicalDevice that pname:device was created from.
```

Comment:
If I may direct your attention, I wasn't sure about this VU:
 - not sure if it matches the intent of the original VU below
 - not sure if it is not handled by `VK_ERROR_OUT_OF_DEVICE_MEMORY` instead

Reply:
I do think it matches the intent of the removed VU.

I also think it is subtly different to the error message and worth preserving, since removing it might constitute a (albeit minor) functionality change.

Reply:
(It's *possible* that someone has set their allocator up to only handle allocations of sizes less than that, and runs into undefined territory due to e.g. integer overflows - unlikely - but technically possible)